src/bucket/BucketList.cpp

```diff
@@ -410,18 +413,31 @@ BucketList::addBatch(Application& app, uint32_t currLedger,
             //           << " to level " << i;
 
             mLevels[i].commit();
-            mLevels[i].prepare(app, currLedger, snap, shadows);
+            mLevels[i].prepare(app, currLedger, currLedgerProtocol, snap,
+                               shadows);
         }
     }
 
     assert(shadows.size() == 0);
-    mLevels[0].prepare(
-        app, currLedger,
-        Bucket::fresh(app.getBucketManager(), liveEntries, deadEntries),
-        shadows);
+    mLevels[0].prepare(app, currLedger, currLedgerProtocol,
+                       Bucket::fresh(app.getBucketManager(),
+                                     currLedgerProtocol,
+                                     initEntries, liveEntries, deadEntries),
+                       shadows);
     mLevels[0].commit();
 }
 
+void
+BucketList::addBatch(Application& app, uint32_t currLedger,
```

Comment:
I would remove this method as I *think* the only legitimate use of that is test code (and we should just have test helper methods elsewhere)

Reply:
done.